packages/form-core/src/FormApi.ts

```diff
@@ -875,7 +895,28 @@ export class FormApi<
                 }))
               }
             }
+          } else {
```

Comment:
Comments for sync also apply here, but let's make sync work first, then we'll replicate in the async :)